releaser/main.go

```diff
@@ -203,7 +210,13 @@ func (r *Releaser) Publish(
 	report.Artifacts = append(report.Artifacts, artifact)
 
 	if report.hasErrors() {
-		// early-exit if engine or cli could not publish
+		// early-exit if engine / cli could not Publish
+		return &report, nil
+	}
+
+	isPrerelease := semver.IsValid(version) && semver.Prerelease(version) != ""
+	if isPrerelease {
```

Comment:
I would be tempted to wrap this in a `isPrerelease()` function, and just reuse it in both places, but just a nit, not worth blocking on it.